File: src/TuRuta/TuRuta.Web/ClientApp/clients/StopsClient.ts
export default class StopsClient {
    async GetRoutes(id: string) {
        var response = await fetch('api/stops/getroutes/' + id);
        if (response.ok) {
            var json = await response.json();
            return json as routeVM[];
        }

        return null;
    }

    async Get() {
        var response = await fetch("api/stops");
        if (response.ok) {
            var json = await response.json();
            return json as stopVM[];
        }

        return null;
    }

    async Create(stop: stopVM) {
        var newStop: stopVM = {
            id: "00000000-0000-0000-0000-000000000000",
            name: stop.name,
            location: stop.location
        };
        console.log(JSON.stringify(newStop));

        var response = await fetch("api/stops", {
            method: "POST",
            headers: {
                "Content-Type": "application/json"
            },
            body: JSON.stringify(newStop)
        });
        if (response.ok) {
            var json = await response.json();
            return json as stopVM;
        }

        return null;
    }

    async GetByName(name: string) {
        var response = await fetch('api/stops/' + name);
        if (response.ok) {
            var json = await response.json();
            return json as stopVM;
        }

        return null;
    }

    async Find(hint: string) {
        var response = await fetch('api/stops/find/' + hint);
        if (response.ok) {
            var json = await response.json();
            return json as string[];
        }

        return Array<string>();
    }
}

File: src/TuRuta/TuRuta.Client/Stops/StopsClient.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using TuRuta.Common.ViewModels;

namespace TuRuta.Client.Stops
{
    class StopsClient
    {
		public StopsClient(string baseAddress = null)
		{
			var address = baseAddress ?? "http://localhost:56340/";
			HttpClient.BaseAddress = new Uri(address);
		}

		private HttpClient HttpClient { get; } = new HttpClient();
		public async Task<StopVM> Get(string name)
		{
			var response = await HttpClient.GetAsync($"/api/Stops/{name}");
			if (response.IsSuccessStatusCode)
			{
				return JsonConvert.DeserializeObject<StopVM>(await response.Content.ReadAsStringAsync());
			}
			return default(StopVM);
		}
		public async Task<IEnumerable<string>> Find(string hint)
		{
			var response = await HttpClient.GetAsync($"/api/Stops/Find/{hint}");
			if (response.IsSuccessStatusCode)
			{
				return JsonConvert.DeserializeObject<IEnumerable<string>>(await response.Content.ReadAsStringAsync());
			}
			return default(IEnumerable<string>);
		}
		public async Task<RouteVM> GetRoutes(string id)
		{
			var response = await HttpClient.GetAsync($"/api/Stops/GetRoutes/{id}");
			if (response.IsSuccessStatusCode)
			{
				return JsonConvert.DeserializeObject<RouteVM>(await response.Content.ReadAsStringAsync());
			}
			return default(RouteVM);
		}
	}
}
